### PR TITLE
Add ability to read comma-separated colormaps during enhancement

### PR DESCRIPTION
**Pull request by djhoese**
Currently only binary `.npy` files are supported for reusable colormaps (rather than literal lists in the YAML). This PR allows for parsing of CSV (comma-separated) text files in a similar manner.

Note: Initially this PR only consists of refactoring of some of the enhancement tests.

 - [ ] Closes #xxxx 
 - [x] Tests added 
 - [x] Fully documented

### CODE REVIEW

**Comment by djhoese** [comment]
This is ready for review. It now includes the ability to have colormap files specified relative to the SATPY_CONFIG_PATH.

File: satpy/enhancements/__init__.py [hunk 1]
```diff
@@ -354,10 +356,21 @@ def create_colormap(palette):
 
     **From a file**
 
-    Colormaps can be loaded from ``.npy`` files as 2D raw arrays with rows for
-    each color. The filename to load can be provided with the ``filename`` key
-    in the provided palette information. The colormap is interpreted as 1 of 4
-    different "colormap modes": ``RGB``, ``RGBA``, ``VRGB``, or ``VRGBA``. The
+    Colormaps can be loaded from ``.npy``, ``.npz``, or comma-separate text
```
Comment (pnuu) [inline satpy/enhancements/__init__.py:359]:
```suggestion
    Colormaps can be loaded from ``.npy``, ``.npz``, or comma-separated text
```

**Review by pnuu — APPROVED** [review]
LGTM. Just one typo.

**Review by mraspaud — COMMENTED** [review]
Looks good, just a couple of minor naming issues.

**Pull request merged**